integration-test-groups/aws2/aws2-cw/src/main/java/org/apache/camel/quarkus/component/aws2/cw/it/ClientBuilder.java

```diff
@@ -0,0 +1,875 @@
+import software.amazon.awssdk.services.cloudwatch.model.PutManagedInsightRulesRequest;
+import software.amazon.awssdk.services.cloudwatch.model.PutManagedInsightRulesResponse;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricAlarmRequest;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricAlarmResponse;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricDataRequest;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricDataResponse;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricStreamRequest;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricStreamResponse;
+import software.amazon.awssdk.services.cloudwatch.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.cloudwatch.model.SetAlarmStateRequest;
+import software.amazon.awssdk.services.cloudwatch.model.SetAlarmStateResponse;
+import software.amazon.awssdk.services.cloudwatch.model.StartMetricStreamsRequest;
+import software.amazon.awssdk.services.cloudwatch.model.StartMetricStreamsResponse;
+import software.amazon.awssdk.services.cloudwatch.model.StopMetricStreamsRequest;
+import software.amazon.awssdk.services.cloudwatch.model.StopMetricStreamsResponse;
+import software.amazon.awssdk.services.cloudwatch.model.TagResourceRequest;
+import software.amazon.awssdk.services.cloudwatch.model.TagResourceResponse;
+import software.amazon.awssdk.services.cloudwatch.model.UntagResourceRequest;
+import software.amazon.awssdk.services.cloudwatch.model.UntagResourceResponse;
+import software.amazon.awssdk.services.cloudwatch.paginators.DescribeAlarmHistoryIterable;
+import software.amazon.awssdk.services.cloudwatch.paginators.DescribeAlarmsIterable;
+import software.amazon.awssdk.services.cloudwatch.paginators.DescribeAnomalyDetectorsIterable;
+import software.amazon.awssdk.services.cloudwatch.paginators.DescribeInsightRulesIterable;
+import software.amazon.awssdk.services.cloudwatch.paginators.GetMetricDataIterable;
+import software.amazon.awssdk.services.cloudwatch.paginators.ListDashboardsIterable;
+import software.amazon.awssdk.services.cloudwatch.paginators.ListManagedInsightRulesIterable;
+import software.amazon.awssdk.services.cloudwatch.paginators.ListMetricStreamsIterable;
+import software.amazon.awssdk.services.cloudwatch.paginators.ListMetricsIterable;
+import software.amazon.awssdk.services.cloudwatch.waiters.CloudWatchWaiter;
+
+public class ClientBuilder {
+
+    @Inject
+    CamelContext camelContext;
+
+    @Produces
+    @Named("customClient")
+    CloudWatchClient produceClient() {
+
+        return new MockedClient();
+    }
+
+    @Produces
+    @Named("customClient2")
+    CloudWatchClient produceClient2() {
+        //second client is required to avoid autowiring of the first client if there is only one client
+        return new MockedClient();
+    }
+
+    class MockedClient implements CloudWatchClient {
```

Comment:
Do we need to re-implement the default methods ?

In Camel, it looks possible not do to so:
https://github.com/apache/camel/blob/main/components/camel-aws/camel-aws2-cw/src/test/java/org/apache/camel/component/aws2/cw/CloudWatchClientMock.java